examples/counter/shared/src/app/mod.rs

```diff
@@ -0,0 +1,172 @@
+use crux_core::render::Render;
+use crux_http::Http;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+const API_URL: &str = "https://crux-counter.fly.dev";
+
+#[derive(Default)]
+pub struct App;
+
+impl crux_core::App for App {
+    type Model = Model;
+    type Event = Event;
+    type ViewModel = ViewModel;
+    type Capabilities = Capabilities;
+
+    fn update(&self, msg: Self::Event, model: &mut Self::Model, caps: &Self::Capabilities) {
+        match msg {
+            Event::Get => {
+                caps.http.get_json(Url::parse(API_URL).unwrap(), Event::Set);
+            }
+            Event::Set(counter) => {
+                model.count = counter.value;
+                caps.render.render();
+            }
+            Event::Increment => {
```

Comment:
We could do optimistic state updates here too (maybe with an extra bit of state indicating whether the value is pending confirmation from the server or something). It would make for a better incremental example in the docs, maybe, starting with a local one, then adding in the remote API on top.

Reply:
Alternatively, some state keeping track of the fact that we've sent something and are awaiting server response, so feedback can be given in the UI.

Reply:
yeah, like the optimistic update idea

Reply:
Although our API is sooo fast you wouldn't see it anyway